http-server/src/main/java/com/facebook/airlift/http/server/HttpServer.java

```diff
@@ -631,4 +645,14 @@ private static ServerConnector createServerConnector(
         connector.open(channel);
         return connector;
     }
+
+    private static Optional<Set<String>> getRolesFromClassMetadata(Servlet servlet)
+    {
+        if (servlet.getClass().isAnnotationPresent(RolesAllowed.class)) {
+            return Optional.of(ImmutableSet.copyOf(servlet.getClass().getAnnotation(RolesAllowed.class).value()));
+        }
+        else {
+            return Optional.empty();
+        }
+    }
```

Comment:
Should we move this logic inside `SecureServlet` ?

Reply:
Sounds reasonable.